File: algorithms/1_easy/B/00868_binary_gap/solution.cpp
class Solution {
public:
    static int binaryGap(const int n_int) {
        uint32_t n = static_cast<uint32_t>(n_int);
        while (n > 0 && (n & 1) != 1) {
            n >>= 1;
        }

        int cnt = 0;
        int max = 0;

        while (n > 1) {
            if ((n & 1) == 1) {
                cnt = 1;
            } else {
                cnt += 1;
            }

            max = std::max(max, cnt);
            n >>= 1;
        }

        return max;
    }
};

File: algorithms/1_easy/B/00868_binary_gap/solution.py
class Solution:
    def binaryGap(self, n: int) -> int:
        while n > 0 and (n & 1) != 1:
            n >>= 1

        cnt = 0
        res = 0

        while n > 1:
            if (n & 1) == 1:
                cnt = 1
            else:
                cnt += 1

            res = max(res, cnt)
            n >>= 1

        return res
